### PR TITLE
[ARCTIC-1759]Change the project name in `README.md` and poms

### LINKED ISSUE #1759: [Subtask]: Change `README.md` and poms for name name `Amoro`
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### Description

As the project will rename to Amoro, the `README.md` and poms should be modified.

### Parent issue

#1744 

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhoujinsong**
## Why are the changes needed?

resolve #1759 

## Brief change log

- Change content in `README.md` to adapt to the new project name.
- Change group-id and artifact-id in poms. 

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Review by shidayang — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -20,14 +20,14 @@
...
-    <groupId>com.netease.arctic</groupId>
-    <artifactId>arctic-parent</artifactId>
+    <groupId>com.netease.amoro</groupId>
+    <artifactId>amoro-parent</artifactId>
     <version>0.5.0-SNAPSHOT</version>
     <packaging>pom</packaging>
 
-    <name>Arctic Project Parent</name>
+    <name>Amoro Project Parent</name>
     <description>Arctic is a LakeHouse management service for open architecture</description>
```
Comment (wangtaohz) [inline pom.xml:29]: It should be `Amoro` instead of `Arctic`

**Review by wangtaohz — COMMENTED** [review]
I left some comments.

File: pom.xml (hunk 1 above)
Comment (zhoujinsong) [reply pom.xml]: yes, i will fix it.

**Review by majin1102 — APPROVED** [review]
LGTM

**Pull request merged**